task_manager/task_manager.cc

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "task_manager.hh"
+#include "db/config.hh"
+
+seastar::gate& task_manager::async_gate() {
+    return _gate;
+}
+
+abort_source& task_manager::abort_source() {
+    return _as;
+}
```

Comment:
We usually place trivial accessors in the header file, and specified as noexcept (plus const where appropriate)